tests/queries/tests.py

```diff
@@ -82,6 +82,86 @@ def setUpTestData(cls):
+            Tag.objects.all().select_for_update().filter(name='test'),
+        ]
+        supported_formats = connection.features.supported_explain_formats
+        all_formats = (None,) + tuple(supported_formats) + tuple(f.lower() for f in supported_formats)
+        for idx, queryset in enumerate(querysets):
+            for format in all_formats:
+                with self.subTest(format=format, queryset=idx):
+                    result = queryset.explain(format=format)
+                    self.assertIsInstance(result, str)
+                    self.assertTrue(result)
+
+    def test_explain_unknown_format(self):
+        with self.assertRaises(ValueError) as exc:
+            Tag.objects.all().explain(format='does not exist')
+
+        self.assertIn('DOES NOT EXIST is not a recognised format', str(exc.exception))
+
+        if connection.features.supported_explain_formats:
+            expected_message = 'Allowed formats: {0}'.format(', '.join(connection.features.supported_explain_formats))
+            self.assertIn(expected_message, str(exc.exception))
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific")
+    def test_postgres_explain_options(self):
+        qs = Tag.objects.filter(name='test').all()
+
+        test_options = [
+            {'COSTS': False, 'BUFFERS': True, 'ANALYZE': True},
+            {'costs': False, 'buffers': True, 'analyze': True},
+            {'verbose': True, 'timing': True, 'analyze': True},
+            {'verbose': False, 'timing': False, 'analyze': True},
+        ]
+
+        if connection.pg_version >= 100000:
+            test_options.append({'summary': True})
+
+        for options in test_options:
+            with self.subTest(**options), transaction.atomic():
+                with CaptureQueriesContext(connection) as captured_queries:
+                    qs.explain(format='text', **options)
+
+                self.assertEqual(len(captured_queries), 1)
+                for name, value in options.items():
+                    name = name.upper()
+                    option = '{0} {1}'.format(name, 'true' if value else 'false')
+                    self.assertIn(option, captured_queries[0]['sql'])
+
+    @unittest.skipUnless(connection.vendor == 'mysql', "Mysql specific")
+    def test_mysql_explain_text_to_traditional(self):
+        qs = Tag.objects.filter(name='test').all()
+
+        with CaptureQueriesContext(connection) as captured_queries:
+            qs.explain(format='text')
+
+        self.assertEqual(len(captured_queries), 1)
+        self.assertIn('FORMAT=TRADITIONAL', captured_queries[0]['sql'])
+
+    @unittest.skipUnless(connection.vendor == 'mysql' and connection.mysql_version < (5, 7), "Mysql < 5.7 specific")
+    def test_mysql_explain_extended(self):
+        qs = Tag.objects.filter(name='test').all()
+
+        with CaptureQueriesContext(connection) as captured_queries:
```

Comment:
Wouldn't `with self.assertNumQueries(1):` also work?

Reply:
perhaps, I wanted to test the contents of the executed query though, is this possible?

Reply:
I don't think so. I hadn't noticed that part.